modules/delegates/api/fetchDelegatesExecSupport.ts

```diff
@@ -0,0 +1,66 @@
+import { gqlRequest } from 'modules/gql/gqlRequest';
+import { allDelegates } from 'modules/gql/queries/allDelegates';
+import { getContracts } from 'modules/web3/helpers/getContracts';
+import { Query } from 'modules/gql/generated/graphql';
+import { allDelegatesExecSupportKey } from 'modules/cache/constants/cache-keys';
+import { cacheGet, cacheSet } from 'modules/cache/cache';
+import { SupportedNetworks } from 'modules/web3/constants/networks';
+import { networkNameToChainId } from 'modules/web3/helpers/chain';
+import logger from 'lib/logger';
+import { ZERO_SLATE_HASH } from 'modules/executive/helpers/zeroSlateHash';
+import { getSlateAddresses } from 'modules/executive/helpers/getSlateAddresses';
+import { DelegateExecSupport } from '../types';
+import { TEN_MINUTES_IN_MS } from 'modules/app/constants/time';
+
+export async function fetchDelegatesExecSupport(network: SupportedNetworks): Promise<{
+  error: boolean;
+  data?: DelegateExecSupport[];
+}> {
+  const cachedResponse = await cacheGet(allDelegatesExecSupportKey, network);
+  if (cachedResponse) {
+    return {
+      error: false,
+      data: JSON.parse(cachedResponse)
+    };
+  }
+
+  try {
+    const chainId = networkNameToChainId(network);
+
+    const data = await gqlRequest<Query>({ chainId, query: allDelegates });
+    const delegates = data.allDelegates.nodes;
+
+    const contracts = getContracts(chainId, undefined, undefined, true);
+    const delegatesExecSupport = await Promise.all(
+      delegates.map(async delegate => {
```

Comment:
You were saying it takes a while to fetch the exec support for all the delegates?  But you think this is better than only passing in the subset of delegates we care about? Alternatively, I wonder if it'd be fine to only have the UI show the exec support for recognized delegates on the main delegates page